L1Trigger/L1TGlobal/plugins/GtExternalFakeProducer.cc

```diff
@@ -0,0 +1,180 @@
+
+    // Setup parameters
+    bxFirst_ = iConfig.getParameter<int>("bxFirst");
+    bxLast_  = iConfig.getParameter<int>("bxLast");
+
+    setBptxAND_ = iConfig.getParameter<bool>("setBptxAND");
+    setBptxPlus_ = iConfig.getParameter<bool>("setBptxPlus");
+    setBptxMinus_ = iConfig.getParameter<bool>("setBptxMinus");
+    setBptxOR_ = iConfig.getParameter<bool>("setBptxOR");
+
+    // set cache id to zero, will be set at first beginRun:
+    m_paramsCacheId = 0;
+  }
+
+
+  GtExternalFakeProducer::~GtExternalFakeProducer()
+  {
+  }
+
+
+
+  //
+  // member functions
+  //
+
+  // ------------ method called to produce the data ------------
+  void
+  GtExternalFakeProducer::produce(Event& iEvent, const EventSetup& iSetup)
+  {
+
+    LogDebug("GtExternalFakeProducer") << "GtExternalFakeProducer::produce function called...\n";
+
+    // Setup vectors
+    GlobalExtBlk extCond_bx;
+
+    // Set the range of BX....TO DO...move to Params or determine from param set.
+    int bxFirst = bxFirst_;
+    int bxLast  = bxLast_;
+
+
+    //outputs
+    std::auto_ptr<GlobalExtBlkBxCollection> extCond( new GlobalExtBlkBxCollection(0,bxFirst,bxLast));
+
+
+    extCond_bx.reset();
+    // Fill in some external conditions for testing
+    if( setBptxAND_ ) extCond_bx.setExternalDecision(8,true);  //EXT_BPTX_plus_AND_minus.v0
+    if( setBptxPlus_ ) extCond_bx.setExternalDecision(9,true);  //EXT_BPTX_plus.v0
+    if( setBptxMinus_ ) extCond_bx.setExternalDecision(10,true); //EXT_BPTX_minus.v0
+    if( setBptxOR_ ) extCond_bx.setExternalDecision(11,true); //EXT_BPTX_plus_OR_minus.v0
+
+    // Fill Externals
+    extCond->push_back(-2, extCond_bx);
+    extCond->push_back(-1, extCond_bx);
+    extCond->push_back(0,  extCond_bx);
+    extCond->push_back(1,  extCond_bx);
+    extCond->push_back(2,  extCond_bx); 
+   
+
+    iEvent.put(extCond);
+
+  }
+
+  // ------------ method called once each job just before starting event loop ------------
+  void
+  GtExternalFakeProducer::beginJob()
+  {
+  }
+
+  // ------------ method called once each job just after ending the event loop ------------
+  void
+  GtExternalFakeProducer::endJob() {
+  }
+
+  // ------------ method called when starting to processes a run ------------
+
+  void GtExternalFakeProducer::beginRun(Run const&iR, EventSetup const&iE){
+
+    LogDebug("GtExternalFakeProducer") << "GtExternalFakeProducer::beginRun function called...\n";
+
+  }
+
+  // ------------ method called when ending the processing of a run ------------
+  void GtExternalFakeProducer::endRun(Run const& iR, EventSetup const& iE){
+
+  }
+
+  // ------------ method fills 'descriptions' with the allowed parameters for the module ------------
+  void
+  GtExternalFakeProducer::fillDescriptions(ConfigurationDescriptions& descriptions) {
+    //The following says we do not know what parameters are allowed so do no validation
+    // Please change this to state exactly what you do use, even if it is no parameters
```

Comment:
Could you actually implement a meaningful  fillDescriptions method ?
If you have a working cfi file, `edmPythonConfigToCppValidation` can write the C++ for you.